expr/func_groupbytags.go

```diff
@@ -57,7 +57,9 @@ func (s *FuncGroupByTags) Exec(cache map[Req][]models.Series) ([]models.Series,
 		if tag == "name" {
 			// We handle name explicitly, remove it from tags
 			useName = true
-			groupTags = append(groupTags[:i], groupTags[i+1:]...)
+			groupTags = make([]string, len(s.tags)-1)
```

Comment:
This is an interesting case, since it's not modifying the input series but the Func itself. It seems like it should actually be moved to a pre-processing step (and maybe out of Exec)? Although, we don't currently reuse Func, so maybe it's not a big deal.

Reply:
oh you are right. we are modifying the func's tag property, not that of an incoming series.
and we are currently not reusing funcs, though there's a small chance that some day we will.

I think we should:
* not merge this PR as is
* restore the previous code
* comment on top that we modify s.tags during Exec
* perhaps even get rid of the seperate groupTags var,and just use s.tags everywhere.